pkgs/development/compilers/zig/default.nix

```diff
@@ -59,16 +38,20 @@ stdenv.mkDerivation rec {
     export HOME=$TMPDIR;
   '';
 
+  patchPhase = ''
+    # Zig's build looks at /usr/bin/env to find dynamic linking info. This
+    # doesn't work in Nix' sandbox. Use env from our coreutils instead.
+    substituteInPlace lib/std/zig/system/NativeTargetInfo.zig --replace "/usr/bin/env" "${coreutils}/bin/env"
+  '';
+
   cmakeFlags = [
     # file RPATH_CHANGE could not write new RPATH
     "-DCMAKE_SKIP_BUILD_RPATH=ON"
   ];
 
   doCheck = true;
-  checkPhase = ''
```

Comment:
Why are you changing from check to installCheck?

Reply:
I don't know where `make` puts the stage3 build. It seems to be in zig-cache. But `make install` puts the stage3 build in a consistent place, so I decided to use installCheck.